app/stores/ada/YoroiTransferStore.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import { observable, action, runInAction } from 'mobx';
+import { restoreWalletForTransfer } from '../../api/ada';
+import { getReceiverAddress } from '../../api/ada/lib/storage/adaAddress';
+import { generateTransferTx } from '../../api/ada/daedalusTransfer';
+import Store from '../base/Store';
+
+import { RustModule } from '../../api/ada/lib/cardanoCrypto/rustLoader';
+
+export default class YoroiTransferStore extends Store {
+  @observable status: TransferStatus = 'gettingMnemonics';
+  @observable transferFundsRequest: Request<TransferFundsFunc>
+    = new Request<TransferFundsFunc>(this._transferFundsRequest);
+  @observable restoreForTransferRequest: Request<RestoreWalletForTransferFunc>
+    = new Request(this.api.ada.restoreWalletForTransfer);
+
+  setup(): void {
+    const actions = this.actions.ada.yoroiTransfer;
+    actions.setupTransferFundsWithMnemonic.listen(this._setupTransferFundsWithMnemonic);
+    actions.backToUninitialized.listen(this._backToUninitialized);
+    actions.transferFunds.listen(this._transferFunds);
+    actions.cancelTransferFunds.listen(this._reset);
+  }
+
+  teardown(): void {
+    super.teardown();
+    this._reset();
+  }
+
+  _setupTransferFundsWithMnemonic = async (payload: { recoveryPhrase: string }): Promise<void> => {
+    this._updateStatus('checkingAddresses');
+    // addresses and indices are one-to-one correspondences
+    const { wallet, addresses, indices } = this._restoreWalletForTransfer(recoveryPhrase);
+    const masterKey = getWalletMasterKey(wallet);
```

Comment:
`getWalletMasterKey` fetches from storage but we don't have this recovery phrase in storage so it won't work. You will need to instead use the Rust library to generate the key in memory.
See how we do this here: https://github.com/input-output-hk/js-cardano-wasm/blob/master/cardano-wallet/tests/index.js#L18